[Data Views] Show loading indicators when submitting a data view form to prevent multiple requests

jughosta opened the pull request:
Closes https://github.com/elastic/kibana/issues/146125

This PR adds loading indicators to the form buttons and also disables them when submitting the form.

* Creating an ad-hoc data view
![Feb-08-2023 16-00-20](https://user-images.githubusercontent.com/1415710/217567470-53e0c614-d12d-4849-90e1-f157c6c36e79.gif)

* Editing an ad-hoc data view
![Feb-08-2023 16-01-38](https://user-images.githubusercontent.com/1415710/217567599-0066393c-4bae-4364-a568-b54c6509d8ad.gif)

* Creating a persisted data view
![Feb-08-2023 16-00-57](https://user-images.githubusercontent.com/1415710/217567411-67dd1b3c-99b9-455b-adbe-be21d61344ee.gif)

* Editing a persisted data view
![Feb-08-2023 16-02-08](https://user-images.githubusercontent.com/1415710/217567541-a6314611-544f-45f2-944d-f49c7a654a43.gif)




<!--ONMERGE {"backportTargets":["8.7"]} ONMERGE-->

elasticmachine commented:
Pinging @elastic/kibana-data-discovery (Team:DataDiscovery)

davismcphee reviewed (commented):
Pulled and tested locally, and everything seems to be working as expected. But I think it would be good to make a quick update to the functional tests to check for the disabled button so that we have some test coverage on it.

jughosta commented:
Hi @mattkime,
Where do we store tests for Data View Editor?

mattkime commented:
@jughosta - `test/functional/apps/management`

jughosta commented:
@davismcphee While working on tests found another issue: when the confirmation modal was shown the form had already finished "submitting" and had reenabled the button. So I fixed that via `await`s and added a test.

<img width="756" alt="Screenshot 2023-02-15 at 11 24 37" src="https://user-images.githubusercontent.com/1415710/219002216-1389b97e-bcd6-4cc7-a212-31538ec4a109.png">

davismcphee reviewed (approved):
> While working on tests found another issue: when the confirmation modal was shown the form had already finished "submitting" and had reenabled the button. So I fixed that via `await`s and added a test.

Good catch! Thanks for fixing and adding the functional test, LGTM!

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.7|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/151407">](https://github.com/elastic/kibana/pull/151407)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->
